WooCommerce/Classes/ViewRelated/InAppFeedback/InAppFeedbackCardViewModel.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+import StoreKit
+
+struct InAppFeedbackCardViewModel {
+    enum Feedback {
+        case liked
+        case didntLike
+    }
+
+    private let analytics: Analytics
+
+    /// Closure invoked after the user has chosen what kind feedback to give.
+    var onFeedbackGiven: ((Feedback) -> Void)?
+
+    private let storeReviewControllerType: SKStoreReviewControllerProtocol.Type
+
+    init(storeReviewControllerType: SKStoreReviewControllerProtocol.Type = SKStoreReviewController.self,
+         analytics: Analytics = ServiceLocator.analytics) {
+        self.storeReviewControllerType = storeReviewControllerType
+        self.analytics = analytics
+    }
+
+    func didTapCouldBeBetter() {
+        analytics.track(event: .appFeedbackPrompt(action: .didntLike))
+        onFeedbackGiven?(.didntLike)
+    }
+
+    func didTapILikeIt() {
+        analytics.track(event: .appFeedbackPrompt(action: .liked))
+        if let windowScene = UIApplication.shared.connectedScenes.first(where: { $0.activationState == .foregroundActive}) as? UIWindowScene {
+            storeReviewControllerType.requestReview(in: windowScene)
```

Comment:
app rating seems to only be called on "I Like It" and never on "Could be better" and I'm not sure what's the correct behavior, based on the test instruction this seems to be called by "Could be better" too.

Reply:
Thanks, Hafiz! I updated the test instructions for the app ratings flow.